### PR TITLE
Bugfix for #840 - Local memory fix

### LINKED ISSUE #823: JSON Decoder Error every single time after a couple of runs
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

After the Auto GPT runs a couple of times (seems to vary but somewhere between y -10 and y -20) it just breaks and gives this error:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/openai/api_requestor.py", line 672, in _interpret_response_line
    data = json.loads(rbody)
           ^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/./Auto-GPT-master/scripts/main.py", line 330, in <module>
    assistant_reply = chat.chat_with_ai(
                      ^^^^^^^^^^^^^^^^^^
  File "/Users/./Auto-GPT-master/scripts/chat.py", line 128, in chat_with_ai
    assistant_reply = create_chat_completion(
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/./Auto-GPT-master/scripts/llm_utils.py", line 19, in create_chat_completion
    response = openai.ChatCompletion.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This

### Current behavior 😯

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/openai/api_requestor.py", line 672, in _interpret_response_line
    data = json.loads(rbody)
           ^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/./Auto-GPT-master/scripts/main.py", line 330, in <module>
    assistant_reply = chat.chat_with_ai(
                      ^^^^^^^^^^^^^^^^^^
  File "/Users/./Auto-GPT-master/scripts/chat.py", line 128, in chat_with_ai
    assistant_reply = create_chat_completion(
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/./Auto-GPT-master/scripts/llm_utils.py", line 19, in create_chat_completion
    response = openai.ChatCompletion.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

### Expected behavior 🤔

It should keep trying instead of breaking.

### Your prompt 📝

```yaml
it's been happening for multiple prompts
```

### LINKED ISSUE #840: New Memory Error
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

I ran AutoGPT successfully yesterday and it seemed to take on the tasks that I asked very well.  I am trying to create an AI specific to a global biomedical project that is about 8 years old, and has GitHub and Forum assets.

### Current behavior 😯

Today I started again, and this is was the output:

File "/home/acumenus/Auto-GPT/scripts/main.py", line 314, in <module>
    memory = get_memory(cfg, init=True)
  File "/home/acumenus/Auto-GPT/scripts/memory/__init__.py", line 33, in get_memory
    memory = LocalCache(cfg)
  File "/home/acumenus/Auto-GPT/scripts/config.py", line 19, in __call__
    cls._instances[cls] = super(
  File "/home/acumenus/Auto-GPT/scripts/memory/local.py", line 32, in __init__
    loaded = orjson.loads(f.read())
orjson.JSONDecodeError: unexpected end of data: line 1 column 1253377 (char 1253376)

### Expected behavior 🤔

_No response_

### Your prompt 📝

Would you like me to return to being Acu-Test-1?
Continue with the last settings?
Name:  Acu-Test-1
Role:  Test whether AutoGPT can create an AI that can answer questions about OHDSI WebAPI
Goals: ['Read all documentation about OHDSI WebAPI on GitHub', 'Read all documentation about OHDSI ATLAS on GitHub', 'Be able to answer questions about how to install Atlas and WebAPI', 'Read all the posts on the OHDSI Forums and learn what problems and solutions are discussed', 'Be able to provide the best answers to common installation problems']
Continue (y/n): y

### PR DESCRIPTION
**Pull request by chozzz**
### Background
- Fixes issue #840 

### Changes
- Fix #840 
    - Added defensive coding for local memory to load the JSON file that was gitignored at 54101c79973ca5ca8ccd7e1ac59856cb282c57d8
    - Added unit test placeholder (but doesn't know how to run it - sorry, still newb in python)

### PR Quality Checklist
- [x] ~My pull request is atomic and focuses on a single change.~
- [x] I have thouroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] ~I have documented my changes clearly and comprehensively.~
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by BillSchumacher** [comment]
This has nothing to do with that error. This json is being read by the orjson library and that error is benig thrown by the openai library's call to the normal json library.

File: scripts/memory/local.py [hunk 1]
```diff
@@ -28,10 +28,20 @@ class LocalCache(MemoryProviderSingleton):
     def __init__(self, cfg) -> None:
         self.filename = f"{cfg.memory_index}.json"
         if os.path.exists(self.filename):
-            with open(self.filename, 'rb') as f:
-                loaded = orjson.loads(f.read())
-                self.data = CacheContent(**loaded)
+            try:
+                with open(self.filename, 'w+b') as f:
```
Comment (BillSchumacher) [inline scripts/memory/local.py:32]: The try statement is probably ok but the rest of this is not needed.

**Review by BillSchumacher — CHANGES_REQUESTED** [review]

**Comment by chozzz** [comment]
@BillSchumacher I referenced the wrong issue. Thanks for pointing that out. I updated the commit message and PR details.

**Comment by BillSchumacher** [comment]
Might be worthwhile to add a test that writes a file with tabs and space and another with invalid json, missing a closing brace or something and tries to load it.

**Review by BillSchumacher — APPROVED** [review]

**Review by nponeccop — APPROVED** [review]

**Review by richbeales — APPROVED** [review]

**Pull request merged**